lib/utils/matchesStringOrRegExp.mjs

```diff
@@ -47,39 +49,47 @@ function testAgainstStringOrRegExpOrArray(value, comparison) {
 	return false;
 }
 
+/**
+ * @param {string} value
+ * @param {RegExp} pattern
+ * @returns {MatchResult}
+ */
+function matchValue(value, pattern) {
+	const match = value.match(pattern);
+
+	return match ? { match: value, pattern, substring: match[0] ?? '' } : false;
+}
+
 /**
  * @param {string} value
  * @param {string | RegExp} comparison
+ * @returns {MatchResult}
  */
 function testAgainstStringOrRegExp(value, comparison) {
 	// If it's a RegExp, test directly
 	if (comparison instanceof RegExp) {
-		const match = value.match(comparison);
-
-		return match ? { match: value, pattern: comparison, substring: match[0] || '' } : false;
+		return matchValue(value, comparison);
 	}
 
 	// Check if it's RegExp in a string
-	const firstComparisonChar = comparison[0];
-	const lastComparisonChar = comparison[comparison.length - 1];
-	const secondToLastComparisonChar = comparison[comparison.length - 2];
-
+	const regexFlag = 'i';
 	const comparisonIsRegex =
-		firstComparisonChar === '/' &&
-		(lastComparisonChar === '/' ||
-			(secondToLastComparisonChar === '/' && lastComparisonChar === 'i'));
-
-	const hasCaseInsensitiveFlag = comparisonIsRegex && lastComparisonChar === 'i';
+		comparison.startsWith('/') &&
+		(comparison.endsWith('/') || comparison.endsWith(`/${regexFlag}`));
```

Comment:
```suggestion
		comparison.match(`\/${regexFlag}?$`);
```

Reply:
`endsWith` seems a bit faster than a regex, right?

Reply:
And, I feel `endsWith` is a bit clearer than a regex, considering `startsWith`.